symfony/plugins/orangehrmRecruitmentPlugin/Api/VacancyAPI.php

```diff
@@ -0,0 +1,387 @@
+//        } else {
+//            $vacancyParamHolder->setStatus(0);
+//        }
+
+        $vacancies = $this->getVacancyService()->getAllVacancies($vacancyParamHolder);
+        $count = $this->getVacancyService()->getVacancyDao()->searchVacanciesCount($vacancyParamHolder);
+//        var_dump($count);
+        return new EndpointCollectionResult(
+            VacancyModel::class,
+            $vacancies,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => $count])
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+//            $this->getValidationDecorator()->notRequiredParamRule(
+//                new ParamRule(self::FILTER_VACANCY_ID)
+//            ),
+//            $this->getValidationDecorator()->notRequiredParamRule(
+//                new ParamRule(self::FILTER_HIRING_MANAGER_ID)
+//            ),
+//            $this->getValidationDecorator()->notRequiredParamRule(
+//                new ParamRule(self::FILTER_JOB_TITLE_ID)
+//            ),
+//            $this->getValidationDecorator()->notRequiredParamRule(
+//                new ParamRule(self::FILTER_STATUS)
+//            ),
+            new ParamRule(self::FILTER_VACANCY_ID),
+            new ParamRule(self::FILTER_HIRING_MANAGER_ID),
+            new ParamRule(self::FILTER_JOB_TITLE_ID),
+            new ParamRule(self::FILTER_STATUS),
+
+            ...$this->getSortingAndPaginationParamsRules(VacancySearchFilterParams::ALLOWED_SORT_FIELDS)
+
+        );
+    }
+
+    /**
+     * @inheritDoc
+     * @throws DaoException
+     */
+    public function create(): EndpointResult
+    {
+        $vacancy = new Vacancy();
+        $this->setVacancy($vacancy);
+
+        $vacancy = $this->getVacancyService()->saveJobVacancy($vacancy);
+
+        return new EndpointResourceResult(VacancyModel::class, $vacancy);
+    }
+
+    /**
+     * @param  Vacancy  $vacancy
+     */
+    private function setVacancy(Vacancy $vacancy): void
+    {
+        $vacancy->setName(
+            $this->getRequestParams()->getString(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_NAME
+            )
+        );
+        $vacancy->setDescription(
+            $this->getRequestParams()->getString(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_DESCRIPTION
+            )
+        );
+        $vacancy->setNumOfPositions(
+            $this->getRequestParams()->getInt(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_NUM_OF_POSITIONS
+            )
+        );
+        $vacancy->getDecorator()->setIsPublished(
+            $this->getRequestParams()->getBooleanOrNull(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_IS_PUBLISHED
+            )
+        );
+        $vacancy->setDefinedTime(
+            new DateTime()
```

Comment:
use `DateTimeHelper`
```php
class <>
{
    use DateTimeHelperTrait;

    $vacancy->setDefinedTime($this->getDateTimeHelper()->getNow());
```
this will make easy for unit tests mocking